src/cephadm/cephadmlib/constants.py

```diff
@@ -9,7 +9,7 @@
 DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
 DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
 DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
-DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/ceph-grafana:9.4.12'
+DEFAULT_GRAFANA_IMAGE = 'docker.io/grafana/grafana:9.4.12'
```

Comment:
any chance we can get this mirrored on quay, like we did for haproxy/keepalive images? Would prefer not to be pulling more things from here when possible as it means we need to be concerned with the rate limit.

Reply:
we can do that. do we usually mirror only the image that we want or add some filters and mirror all the stable versions like v9* or v10*?

Reply:
I bumped into [this issue where](https://github.com/grafana/grafana/issues/27677) community is asking for a different registry other than docker because the rate limit concerns but so far there is no update. And there is also some information about them having a [paid dockerhub so rate limiting is not a concern](https://github.com/grafana/grafana/issues/27677#issuecomment-716588275) with grafana docker images. But people are asking for a registry other than docker there

Reply:
For haproxy and keepalive I just got @guits to push the image to the ceph quay account and that's worked fine (although he is out this week). I assume we could do similar here. I don't think we'd be too picky on the versions as long as it isn't changing super often. I guess we only mirror whatever the default is in cephadm, but if we wanted two or three for whatever reason I assume it wouldn't be a big deal. But can't answer that for sure.

Reply:
For the grafana, I think we will need to update it often than the other services because of the constant CVEs that we get. We can check with @guits when he is back and see how we can mirror them.

Reply:
@nizamial09 @adk3798 grafana images are now mirrored from docker.io to quay.io 

see https://quay.io/repository/ceph/grafana?tab=tags

we might want to do the same with `loki`, `promtail`, etc..

Reply:
thank you @guits, I updated the code to pick the grafana image from `quay.io/ceph/grafana`